pkgs/applications/blockchains/lighthouse/default.nix

```diff
@@ -0,0 +1,76 @@
+{ lib, stdenv
+, fetchFromGitHub, fetchgit
+, rustPlatform
+, pkg-config, cmake, protobuf
+, zlib, bzip2, openssl, leveldb
+, nodePackages, curl, gnumake, git
+, portable ? true
+}:
+
+let
+  # Contains the data required by the deposit_contract
+  # See `common/deposit_contract/build.rs` in the lighthous source for exact version
+  eth2specs = fetchFromGitHub {
+    owner = "ethereum";
+    repo = "eth2.0-specs";
+    rev = "v0.12.1";
+    sha256 = "05irxnskcsdbp844hgk2hwkjmgfjxin99jr1fl5b5wg1v7v7gl4y";
+  };
+  eth2specs_unsafe = fetchFromGitHub {
+    owner = "sigp";
+    repo = "unsafe-eth2-deposit-contract";
+    # NOTE: the version of the unsafe contract lags the main tag, but the v0.9.2.1 code is compatible
+    # with the unmodified v0.12.1 contract. See `common/deposit_contract/build.rs` for details.
+    rev = "v0.9.2.1";
+    sha256 = "151ahdj5114xb5hbhzsx5rx4yyry832hhm0bkx36s2np3r8v883q";
+  };
+in
+rustPlatform.buildRustPackage rec {
+  pname = "lighthouse";
+  version = "1.4.0";
+
+  src = fetchgit {
+    url = "https://github.com/sigp/lighthouse.git";
+    rev = "v${version}";
+    sha256 = "05bp3qx79gkgrbl1mdz8khi16252xkzksll980f3bff2aka8sw04";
+    # Version detection requires git to be present
```

Comment:
Can we patch the command that determines this instead?

Reply:
I tried but it was painful to maintain and I had to rework it for every release. So I gave up and just let it do it's thing.